plugin/tiff/src/main/java/it/geosolutions/imageioimpl/plugins/tiff/TIFFImageReader.java

```diff
@@ -382,9 +401,40 @@ public void setInput(Object input,
                     ("input not an ImageInputStream!"); 
             }
             this.stream = (ImageInputStream)input;
+            // Check for external masks/overviews
+            if (input instanceof FileImageInputStreamExtImpl) {
+
```

Comment:
Not sure... should we do an instanceof on the interface instead (FileImageInputSreamExt)?